syntax/verilog_systemverilog.vim

```diff
@@ -155,6 +154,18 @@ if index(s:verilog_syntax_fold, "function") >= 0 || index(s:verilog_syntax_fold,
 else
     syn keyword verilogStatement  function endfunction
 endif
+if index(s:verilog_syntax_fold, "class") >= 0 || index(s:verilog_syntax_fold, "all") >= 0
+    syn region  verilogFold
+        \ matchgroup=verilogStatement
+        \ start="\(typedef\s\+\)\@<!\(\(virtual|interface\)\s\+\)\?\<class\>"
```

Comment:
Again, I think that you simply need the following:

     start="\(typedef\s\+\)\@<!\<class\>"